src/portal-components/DetailsPage.tsx

```diff
@@ -240,25 +244,41 @@ export default class DetailsPage extends React.Component<
         </div>
       )
     }
-    const synapseConfigContent = (
-      <>
-        {isLoading && <BarLoader color="#878787" loading={true} height={5}/>}
-        {!isLoading && this.renderSynapseConfigArray()}
-      </>
-    )
-    if (showMenu) {
+
+    if (tabLayout?.length) {
       return (
-        <div className="DetailsPage">
-          <div className="button-container">{this.renderMenu()}</div>
+        <div className="DetailsPage tab-layout">
           <div className="component-container" ref={this.ref}>
-            {synapseConfigContent}
+            {<DetailsPageTabs
+              tabConfigs={tabLayout}
+              loading={isLoading}
+              tabContents={!isLoading && this.buildTabContent()}>
+            </DetailsPageTabs>}
```

Comment:
Here I moved the tab code to its own file. Tab settings are from synapse portals configs. Tab contents are passed as an array to the DetailsPageTabs component.